sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -126,7 +126,15 @@ final class Decimal extends Ordered[Decimal] with Serializable {
   def set(decimal: BigDecimal): Decimal = {
     this.decimalVal = decimal
     this.longVal = 0L
-    this._precision = decimal.precision
+    if (decimal.precision <= decimal.scale) {
```

Comment:
But the comment is `// For Decimal, we expect the precision is equal to or large than the scale`.

`=` has been processed within the function `floor` and `ceil`.
<https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala#L387>

This is reason that I think we should use `if (decimal.precision < decimal.scale)`, and it works fine for `0.90`.